code/modules/mob/living/carbon/shock.dm

```diff
@@ -35,13 +70,88 @@
 
 	return traumatic_shock
 
-/mob/living/carbon/human/getPainFromDam()
-	var/value = 0
-	for(var/obj/item/organ/external/organ in organs)
-		value += organ.burn_dam
-		value += organ.brute_dam
-		value *= max((get_specific_organ_efficiency(OP_NERVE, organ.organ_tag)/100), 0.5)
-	return value
-
 /mob/living/carbon/proc/handle_shock()
 	updateshock()
+
+/mob/living/carbon/human/handle_shock()
+	..()
+	if(status_flags & GODMODE)	//godmode
+		return 0
+	if(species && species.flags & NO_PAIN)
+		return
+	if(status_flags & HARDCRIT)	//already in hardcrit
+		return
+
+	//Get crit treshold
+	var/soft_crit_threshold = SOFTCRIT_TRAUMATIC_SHOCK + stats.getStat(STAT_TGH)
+	var/hard_crit_threshold = HARDCRIT_TRAUMATIC_SHOCK + stats.getStat(STAT_TGH)
+	if(stats.getPerk(PERK_BALLS_OF_PLASTEEL))
+		soft_crit_threshold += 20
+		hard_crit_threshold += 20
+
+	//Get shock speed
+	var/shock_stage_speed = 1
+	if(traumatic_shock > soft_crit_threshold + SHOCK_STAGE_BUFFER)
+		shock_stage_speed = 2
+
+	//Handle shock
+	if(shock_stage <= traumatic_shock)	//Shock stage slowly climbs to traumatic shock
+		shock_stage = min(shock_stage + shock_stage_speed, traumatic_shock)
+
+		if(shock_stage < traumatic_shock * 0.4)	//If the difference is too big shock stage jumps to 40% of traumatic shock
+			shock_stage = (traumatic_shock * 0.4)
+
+	else
+		shock_stage = max(shock_stage - shock_stage_speed, 0)
+		return
+
+	if(shock_resist || soft_crit_threshold > traumatic_shock)
+		shock_stage = min(shock_stage, 58)
+
+	sanity.onShock(shock_stage)
+
+	if(shock_stage == 10)
+		to_chat(src, "<span class='danger'>[pick("It hurts so much", "You really need some painkillers", "Dear god, the pain")]!</span>")
+
+	if(shock_stage >= 30)
+		if(shock_stage == 30) emote("me",1,"is having trouble keeping their eyes open.")
+		stuttering = max(stuttering, 5)
+
+	if(shock_stage == 40)
+		to_chat(src, "<span class='danger'>[pick("The pain is excruciating", "Please, just end the pain", "Your whole body is going numb")]!</span>")
+
+	if (shock_stage >= 60)
+		if(shock_stage == 60) emote("me",1,"'s body becomes limp.")
+		if (prob(2))
+			to_chat(src, "<span class='danger'>[pick("The pain is excruciating", "Please, just end the pain", "Your whole body is going numb")]!</span>")
```

Comment:
```suggestion
			to_chat(src, SPAN_DANGER("[pick("The pain is excruciating", "Please, just end the pain", "Your whole body is going numb")]!"))
```

Try to use `SPAN_DANGER` and[ other common span macros](https://github.com/discordia-space/CEV-Eris/blob/master/code/__DEFINES/misc.dm#L224) when possible

Reply:
Span macros can't compile with `pick` inside of them. Look at commit 694f4b6.

Reply:
Shouldn't doing
to_chat(src, SPAN_DANGER(pick("The pain is excruciating!", "Please, just end the pain!", "Your whole body is going numb!")))
work?